PyStemmusScope/iostreamer.py

```diff
@@ -0,0 +1,149 @@
+        creates directory for each forcing data and name it using the name of the
+        station.
+
+        Args:
+            forcing_filenames_list: List of forcing filenames to be used for the experiments.
+            full_run: If true, all stations with forcing listed in the ForcingPath will be used.
+
+        Returns:
+            Dictionary containing path to work directory and config file for every station/forcing.
+        """
+        self.forcing_filenames_list = forcing_filenames_list
+        
+        if full_run:
+            self.forcing_filenames_list = [file.name for file in Path(self.config["ForcingPath"]).iterdir()]
+
+        # dict to store path to config file and work directory for each station
+        self.work_dir_dict, self.config_path_dict = self._create_work_dir()
+
+        return self.work_dir_dict, self.config_path_dict
+
+    def _create_work_dir(self):
+        """Create input directory and copy forcing files.
+
+        Work flow executor to create work directory and all sub-directories.
+
+        Returns:
+            Dictionary containing path to work directory and config file for every station/forcing.
+        """
+        # empty dict to store path to config file and work directory for each station
+        config_path_dict = {}
+        work_dir_dict = {}
+        for ncfile in self.forcing_filenames_list:
+            # get start time with the format Y-M-D-HM
+            timestamp = time.strftime('%Y%m%d_%H%M')
+            station_name = ncfile.split('_')[0]
+            # create input directory
+            work_dir = Path(self.config["InputPath"], station_name + '_' + timestamp)
+            Path(work_dir).mkdir(parents=True, exist_ok=True)
+            print(f"Prepare work directory {work_dir} for the station: {station_name}")
+            # copy model parameters to work directory
+            self._copy_data(work_dir)
+            # update config file for ForcingFileName and InputPath
+            config_file_path = self._update_config_file(ncfile, work_dir, station_name, timestamp)
+            # save config path and work directory to the dictionaries
+            work_dir_dict[ncfile] = work_dir
+            config_path_dict[ncfile] = config_file_path
+
+        return work_dir_dict, config_path_dict
+
+    def _copy_data(self, work_dir):
+        """Copy required data to the work directory.
+
+        Create sub-directories inside the work directory and copy data.
+
+        Args:
+            work_dir: Path to the work directory.
+        """
+        folder_list_vegetation = ["Directional", "FluspectParameters", "Leafangles",
+            "Radiationdata", "SoilSpectra"]
+        for folder in folder_list_vegetation:
+            os.makedirs(work_dir / folder, exist_ok=True)
+            distutils.dir_util.copy_tree(str(self.config[folder]), str(work_dir / folder))
+        
+        # copy input_data.xlsx
+        distutils.file_util.copy_file(str(self.config["InputData"]), str(work_dir))
```

Comment:
```suggestion
        shutil.copy(str(self.config["InputData"]), str(input_dir))
```

Reply:
Same as above. Since we already use `distutils`, it makes more sense to stick to it for copying of single file.